File: clojure/day1_part1_2019.clj

(defn calculate-fuel [mass]
  (- (quot mass 3) 2))

(defn read-input []
  (->> (slurp "input.txt")
       (clojure.string/split-lines)
       (map read-string)))

(defn total-fuel []
  (->> (read-input)
       (map calculate-fuel)
       (reduce +)))

(println (total-fuel))

File: fortran90/day1_part1_2019.f90
program fuel_counter
  implicit none

  integer, parameter :: dp = selected_real_kind(15)
  real(dp) :: total_fuel, fuel, mass
  integer :: io, i

  open(unit=10, file='input.txt', status='old', action='read')
  total_fuel = 0.0_dp

  do
    read(10, *, iostat=io) mass
    if (io /= 0) exit

    fuel = mass / 3.0_dp
    fuel = floor(fuel)
    fuel = fuel - 2.0_dp
    if (fuel < 0.0_dp) fuel = 0.0_dp

    total_fuel = total_fuel + fuel
  end do

  close(10)

  write(*, '(A,F15.2)') 'Total fuel requirement: ', total_fuel

end program fuel_counter